tests/unit/metrics/test_log.py

```diff
@@ -0,0 +1,172 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numbers
+
+import pytest
+
+from gordon.metrics import log
+
+
+@pytest.mark.parametrize('context,context_str', [
+    ({'source': 'external'}, ' context: {\'source\': \'external\'}'),
+    ({}, '')
+])
+def test_logger_adapter_log_outputs_message(caplog, context, context_str):
+    """Metric is formatted and output to log."""
+    logger_adapter = log.LoggerAdapter('INFO')
+    metric = {
+        'metric_name': 'events',
+        'value': 12.12,
+        'context': context
+        }
+    logger_adapter.log(metric)
+    assert 1 == len(caplog.records)
+    expected_log = '[events] value: 12.12' + context_str
+    assert expected_log == caplog.records[0].msg
+
+
+@pytest.fixture
+def mock_logger_adapter(mocker):
+    return mocker.Mock(log.LoggerAdapter)
+
+
+@pytest.fixture
+def log_timer(mock_logger_adapter):
+    metric = {'metric_name': 'test-metric', 'value': 13.05, 'context': {}}
+    return log.LogTimer(metric, mock_logger_adapter, 1)
+
+
+@pytest.mark.asyncio
+async def test_log_timer_manual(log_timer):
+    """Times when using start/stop interface."""
+
+    await log_timer.start()
+    assert isinstance(log_timer._start_time, numbers.Number)
+
+    await log_timer.stop()
+    assert 1 == log_timer.logger.log.call_count
+    actual_metric = log_timer.logger.log.mock_calls[0][1][0]
+    assert 'test-metric' == actual_metric['metric_name']
+    assert {} == actual_metric['context']
+    assert 0 < actual_metric['value']
+
+
+@pytest.mark.asyncio
+async def test_log_timer_as_context_manager(log_timer):
+    """Times when using as a context manager."""
+    async with log_timer as t:
+        assert isinstance(t._start_time, numbers.Number)
+    actual_metric = log_timer.logger.log.mock_calls[0][1][0]
+    assert 1 == log_timer.logger.log.call_count
+    assert 'test-metric' == actual_metric['metric_name']
+    assert {} == actual_metric['context']
+    assert 0 < actual_metric['value']
+
+
+@pytest.fixture
+def relay_config():
+    return {
+        'time_unit': 1,
+        'log_level': 'info',
+    }
+
+
+@pytest.mark.parametrize('name,val,context', [
+    ('a-metric', 12, None),
+    ('nother-metric', 0.0003, {'source': 'internal'}),
+])
+def test_logrelay_create_metric(relay_config, name, val, context):
+    """Create a metric dictionary."""
+    relay = log.LogRelay(relay_config)
+    actual = relay._create_metric(name, val, context=context)
+
+    expected = {
+        'metric_name': name,
+        'value': val,
+        'context': context or {}
+    }
+    assert expected == actual
+
+
+@pytest.fixture
+def mock_log_relay(monkeypatch, mocker, relay_config):
```

Comment:
Could we rename it something like `log_relay` or `log_relay_inst`? Prefacing it with `mock` makes me think the relay itself is mocked out.